python/cugraph/cugraph/cores/core_number.py

```diff
@@ -54,14 +67,52 @@ def core_number(G):
 
     """
 
-    G, isNx = ensure_cugraph_obj_for_nx(G)
+    G, _ = ensure_cugraph_obj_for_nx(G)
+
+    if degree_type is not None:
+        warning_msg = (
+            "The 'degree_type' parameter is ignored in this release.")
+        warnings.warn(warning_msg, Warning)
+
+    if G.is_directed():
+        raise ValueError("input graph must be undirected")
+
+    # FIXME: enable this check once 'degree_type' is supported
+    """
+    if degree_type not in ["incoming", "outgoing", "bidirectional"]:
+        raise ValueError(f"'degree_type' must be either incoming, "
+                         f"outgoing or bidirectional, got: {degree_type}")
+    """
 
-    df = core_number_wrapper.core_number(G)
+    srcs = G.edgelist.edgelist_df['src']
+    dsts = G.edgelist.edgelist_df['dst']
+    weights = G.edgelist.edgelist_df['weights']
+
+    if srcs.dtype != 'int32':
+        raise ValueError(f"Graph vertices must have int32 values, "
+                         f"got: {srcs.dtype}")
+
+    resource_handle = ResourceHandle()
+    graph_props = GraphProperties(
+        is_symmetric=True, is_multigraph=G.is_multigraph())
+    store_transposed = False
+
+    # FIXME:  This should be based on the renumber parameter set when creating
+    # the graph
+    renumber = False
+    do_expensive_check = False
+
+    sg = SGGraph(resource_handle, graph_props, srcs, dsts, weights,
```

Comment:
Actually, I'm confused.  Does this PR also contain some of the commits from PR 2394?  If so then it needs to be merged after and we should go ahead and update this PR to use the copy of the SGGraph (and MGGraph below) that's in G.

Reply:
> Based on https://github.com/rapidsai/cugraph/pull/2394 - should this be removed and use the copy that's in G

Yes, this will be done once #2394 is ready. I got yesterday's change but Alex is about to push more changes. I would like to tests those changes before enabling core-number to use PLC graph

> This would suggest either merging PR 2394 first

Right Alex's PR will be merged before mine. I merged her changes to my PR in order to test it on core-number(which uses PLC graph but I haven't pushed those changes yet until #2394 is ready)

Reply:
> Actually, I'm confused. Does this PR also contain some of the commits from PR 2394? If so then it needs to be merged after and we should go ahead and update this PR to use the copy of the SGGraph (and MGGraph below) that's in G.

Right. I will push those changes once #2394 is ready. to give you more context, I need to set parameters that do not exist in #2394(like `legacy_renum_only` when creating the `dask_cudf` from an edgelist) yet, that's why I am waiting for it to be ready.